tests/test_locality_Voronoi.py

```diff
@@ -328,6 +334,26 @@ def test_voronoi_weights_fcc(self):
         )
         npt.assert_allclose(wrapped_distances, vor.nlist.distances)
 
+    def test_diamond(self):
```

Comment:
@b-butler @tommy-waltmann This is an example of a test that fails without this PR. Before this PR, the Voronoi neighbor bonds are defined by a minimum distance and do not distinguish the actual vectors between particles. This fails for many systems that are small unit cells. In this case, the two particles are bonded to each other with several unique vectors that could have different lengths. The bonds between the particle and itself should have a nonzero vector (pointing to a periodic image outside the cell). When computing the distance based on the wrapped i-j vector, we lose information that can be retained if we track the vector when the neighbor is found.

Without this PR, it is likely that small unit cells will give incorrect results for most quantities related to bonding, like PMFTs and Bond Order Diagrams, when the bonds are defined using Voronoi neighbors.

Also, I updated this branch to resolve conflicts with `master`.

Reply:
Okay, this example makes the motivation for this PR more concrete, thanks for the explanation.